src/main/java/uk/gov/di/handlers/JwkHandler.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.di.handlers;
+
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.RSAKey;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JwkHandler implements Route {
+    private final RSAPublicKey publicKey;
+
+    public JwkHandler(RSAPublicKey publicKey) {
+        this.publicKey = publicKey;
+    }
+
+    @Override
+    public Object handle(Request request, Response response) {
+
+        var jwk = new RSAKey.Builder(publicKey).build();
+        List<JWK> keys = new ArrayList<>();
+        response.type("application/json");
+        response.header("Cache-Control", "max-age=86400");
+
+        try {
+            keys.add(JWK.parse(jwk.toString()));
+            JWKSet jwkSet = new JWKSet(keys);
+            return jwkSet.toJSONObject(true);
```

Comment:
How much of this can we do in the constructor? Shouldn't matter now that performance tests no longer use this stub, but it seems a shame to recalculate this on every request

Reply:
That is a very good point, almost all of this could go in the constructor. Seems like a nice enhancement to make, so will add this in